### PR TITLE
Openx: Pass through imp.ext wholesale except prebid, bidder

### PR DESCRIPTION
**Pull request by laurb9**
OpenX adapter: pass all of incoming `imp.ext` through to endpoint, except for `prebid` and `bidder` fields.

This should not impact existing functionality (`customParams` are still merged into `imp.ext`) but allows reading `gpid` and a host of other openrtb `data` of interest server-side while keeping the adapter code simple.

PBS-Go implementation is https://github.com/prebid/prebid-server/pull/2592

### CODE REVIEW

File: src/main/java/org/prebid/server/bidder/openx/OpenxBidder.java [hunk 1]
```diff
@@ -51,6 +49,8 @@ public class OpenxBidder implements Bidder<BidRequest> {
     private static final String OPENX_CONFIG = "hb_pbs_1.0.0";
     private static final String DEFAULT_BID_CURRENCY = "USD";
 
+    private static final List<String> IMP_EXT_SKIP_FIELDS = List.of("bidder", "prebid");
```
Comment (SerhiiNahornyi) [inline src/main/java/org/prebid/server/bidder/openx/OpenxBidder.java:52]: Pls use `Set` instead of `List`

File: src/main/java/org/prebid/server/bidder/openx/OpenxBidder.java [hunk 2]
```diff
@@ -226,10 +226,16 @@ private ExtPrebid<ExtImpPrebid, ExtImpOpenx> parseOpenxExt(Imp imp) {
         return impExt;
     }
 
-    private ObjectNode makeImpExt(Map<String, JsonNode> customParams, ExtImpAuctionEnvironment auctionEnvironment) {
-        return customParams != null || auctionEnvironment != ExtImpAuctionEnvironment.SERVER_SIDE_AUCTION
-                ? mapper.mapper().valueToTree(OpenxImpExt.of(customParams, auctionEnvironment))
-                : null;
+    private ObjectNode makeImpExt(ExtPrebid<ExtImpPrebid, ExtImpOpenx> impExtPrebid, Imp imp) {
```
Comment (SerhiiNahornyi) [inline src/main/java/org/prebid/server/bidder/openx/OpenxBidder.java:229]: What if

```java
 private ObjectNode makeImpExt(ExtPrebid<ExtImpPrebid, ExtImpOpenx> impExtPrebid, ObjectNode impExt) {
        final ObjectNode updatedExt = impExt.deepCopy();
        updatedExt.remove(IMP_EXT_SKIP_FIELDS);
        final Map<String, JsonNode> customParams = impExtPrebid.getBidder().getCustomParams();
        if (MapUtils.isNotEmpty(customParams)) {
            updatedExt.set(CUSTOM_PARAMS_FIELD_NAME, mapper.mapper().valueToTree(customParams));
        }

        return updatedExt;
    }
```

Comment (SerhiiNahornyi) [reply src/main/java/org/prebid/server/bidder/openx/OpenxBidder.java:229]: Then no need of OpenxImpExt class

File: src/test/resources/org/prebid/server/it/openrtb2/openx/test-auction-openx-request.json [hunk 3]
```diff
@@ -19,6 +19,13 @@
               "bar2"
             ]
           }
+        },
+        "gpid": "gpidvalue",
```
Comment (SerhiiNahornyi) [inline src/test/resources/org/prebid/server/it/openrtb2/openx/test-auction-openx-request.json:23]: We are trying to test different data in unit tests, not in integration
Pls remove this

**Review by SerhiiNahornyi — CHANGES_REQUESTED** [review]

File: src/main/java/org/prebid/server/bidder/openx/OpenxBidder.java (hunk 2 above)
Comment (laurb9) [reply src/main/java/org/prebid/server/bidder/openx/OpenxBidder.java]: I agree, if we don't need to keep the OpenxImpExt for exposing the params, it can be simpler.
I pushed a change along these lines that also skips the encoding by copying customParams from imp.ext.bidder, let me know how it looks.

**Comment by laurb9** [comment]
Thank you @SerhiiNahornyi for the prompt review!

File: src/main/java/org/prebid/server/bidder/openx/OpenxBidder.java [hunk 4]
```diff
@@ -119,9 +122,13 @@ private List<BidRequest> makeRequests(BidRequest bidRequest, List<Imp> bannerImp
...
-        return imp.getBanner() != null
-                ? OpenxImpType.banner
-                : imp.getVideo() != null ? OpenxImpType.video : OpenxImpType.other;
+        if (imp.getBanner() != null) {
+            return OpenxImpType.banner;
+        } else if (imp.getVideo() != null) {
+            return OpenxImpType.video;
+        } else {
+            return OpenxImpType.other;
+        }
```
Comment (laurb9) [inline src/main/java/org/prebid/server/bidder/openx/OpenxBidder.java:131]: This unrelated change snuck in to resolve the last sonar warning about nested ternary

**Review by SerhiiNahornyi — APPROVED** [review]
